src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -3752,16 +3759,31 @@ private enum PostSkipAction
                     accessorKind = SyntaxKind.UnknownAccessorDeclaration;
                 }
 
-                BlockSyntax body = null;
+                BlockSyntax blockBody = null;
+                ArrowExpressionClauseSyntax expressionBody = null;
+
                 SyntaxToken semicolon = null;
                 bool currentTokenIsSemicolon = this.CurrentToken.Kind == SyntaxKind.SemicolonToken;
-                if (this.CurrentToken.Kind == SyntaxKind.OpenBraceToken || (validAccName && !currentTokenIsSemicolon && !IsTerminator()))
+                bool currentTokenIsArrow = this.CurrentToken.Kind == SyntaxKind.EqualsGreaterThanToken;
+
+                if (this.CurrentToken.Kind == SyntaxKind.OpenBraceToken
+                            || (validAccName && !currentTokenIsSemicolon && !currentTokenIsArrow && !IsTerminator()))
+                {
+                    blockBody = this.ParseBlock(isMethodBody: true, isAccessorBody: true);
```

Comment:
Here we should have the same logic as in ParseBlockAndExpressionBodiesWithSemicolon around the case when both a block and an expression body are present. Perhaps it makes sense to adjust code inside ParseBlockAndExpressionBodiesWithSemicolon to handle accessors too and call it from this function.